cmd/common.go

```diff
@@ -446,3 +447,47 @@ func connectSSH(ctx context.Context, sshUser, sshIP string, sshPort int, key str
 
 	return client, nil
 }
+
+func copyPastelUpToRemote(ctx context.Context, client *utils.Client, remotePastelUp string) error {
+	// Check if the current os is linux
+	if runtime.GOOS == "linux" {
+		log.WithContext(ctx).Infof("copying pastelup to remote")
+		var localPastelupPath string
+
+		// Get local pastelup path
+		ex, err := os.Executable()
+		if err != nil {
+			return fmt.Errorf("failure in getting path of executable file %s", err)
+		}
+
+		// Check if localPastelupPath is a symlink file
+		if localPastelupPath, err = filepath.EvalSymlinks(ex); err != nil {
+			return fmt.Errorf("local pastelup is symbol link:  %s", err)
+		}
+
+		// Copy pastelup to remote
+		if err := client.Scp(localPastelupPath, remotePastelUp); err != nil {
+			return fmt.Errorf("failure in copying pastelup to remote %s", err)
+		}
+
+	} else {
+		log.WithContext(ctx).Infof("Current OS is not linux, skipping pastelup copy")
+
+		// Download PastelUpExecName from remote and save to remotePastelUp
+		log.WithContext(ctx).Infof("Downloading pastelup from Pastel download portal ...")
+		version := "beta"
+		downloadURL := fmt.Sprintf("%s/%s/%s", constants.DownloadBaseURL, version, constants.PastelUpExecName["Linux"])
+
+		if _, err := client.Cmd(fmt.Sprintf("wget %s -O %s", downloadURL, remotePastelUp)).Output(); err != nil {
+			return fmt.Errorf("failure in downloading pastelup from remote: %s", err.Error())
+		}
+		log.WithContext(ctx).Infof("pastelup downloaded from Pastel download portal")
+	}
+
+	// chmod +x pastelup at remote
+	if _, err := client.Cmd(fmt.Sprintf("chmod +x %s", remotePastelUp)).Output(); err != nil {
```

Comment:
I think this apply for Linux/Unix/Mac only? not for windows?

Reply:
This apply to the remote SuperNode - which is running Linux